aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -68,8 +85,11 @@ const char* SparseCsrTensorImpl::tensorimpl_type_name() const {
 }
 
 void SparseCsrTensorImpl::resize_(int64_t nnz, IntArrayRef size) {
-  auto rows = size[size.size() - 2];
-  auto cols = size[size.size() - 1];
+  TORCH_CHECK((layout_ == kSparseCsr || layout_ == kSparseBsr), "resize_: layout ", layout_, " is not yet supported");  // TODO
+  auto row_index = size.size() - ((layout_ == kSparseCsr || layout_ == kSparseBsr) ? 2 : 1);
```

Comment:
an "SparseCsrTensorImpl::is_transposed" method might be convenient to wrap ```!(layout_ == kSparseCsr || layout_ == kSparseBsr)```. I see 7 invocations of this pattern in this PR.

Reply:
@pearu - that works too. For now it's just a means of reducing the amount of code.

Reply:
Resolved via different layout specific dispatch macros in https://github.com/pytorch/pytorch/pull/75865